src/SFA.DAS.EAS.Portal/Application/Commands/Reservation/AddReservationCommand.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.EAS.Portal.Application.Services;
+using SFA.DAS.EAS.Portal.Client.Database.Models;
+using SFA.DAS.EAS.Portal.Types;
+using SFA.DAS.Reservations.Messages;
+
+namespace SFA.DAS.EAS.Portal.Application.Commands.Reservation
+{
+    public class AddReservationCommand
+    {
+        private readonly IAccountDocumentService _accountsService;
+        private readonly ILogger<AddReservationCommand> _logger;
+
+        public AddReservationCommand(IAccountDocumentService accountsService, ILogger<AddReservationCommand> logger)
+        {
+            _accountsService = accountsService;
+            _logger = logger;
+        }
+
+        public async Task Execute(ReservationCreatedEvent reservedFunding, CancellationToken cancellationToken = default)
+        {
+            _logger.LogInformation("Executing AddReservationCommand");
+
+            var accountDocument = await _accountsService.Get(reservedFunding.AccountId, cancellationToken);
+
+            if (accountDocument == null)
+            {
+                accountDocument = AccountDocument.Create(reservedFunding.AccountId);
+                accountDocument.Account.Name = reservedFunding.AccountLegalEntityName;
```

Comment:
This stores the name of the `AccountLegalEntity` of the first handled `ReservationCreatedEvent` as the name property at the account level. This is then displayed to the user as the legal entity name of the reservation being displayed.

The issues here are
* the reservation shown to the user might not be the reservation first received, in which case we'll show the incorrect legal entity name to the user, for the reservation
* if the account document is created by first handling a `CohortApprovalRequestedCommand`*, then the property looks not to be set and the front end will error with a `NullReferenceException`, when displaying a reservation.

Name does not belong at the account level and should be removed.

* it's not immediately clear that `CohortApprovalRequestedCommand` correctly handles this case. Will investigate.

Reply:
The second issue doesn't exist, as the handler for `CohortApprovalRequestedByProvider` will error if the account hasn't yet been created by an earlier event.